Graphs/Table/index.js

```diff
@@ -218,22 +268,30 @@ class Table extends AbstractGraph {
          * On data change, resetting the paging and filtered data to 1 and false respectively.
          */
         this.resetFilters((currentPage || 1), this.selectedRows);
+        const filterColumns = this.getColumnByContext(columns);
+        const removedColumns = Table.getRemovedColumns(columns, filterColumns, selectedColumns);
+        this.setState({ removedColumns })
+        this.updateData();
+    }
 
-        const removedColumns = [];
-
-        // remove un-selected columns
+    // update column dynamically according to the value passed through the context
+    getColumnByContext(columns) {
+        const { context } = this.props;
+        const filterColumns = []
         columns.forEach((d, index) => {
-            if (selectedColumns && selectedColumns.length) {
-                if (!selectedColumns.find(column => d.label === column)) {
-                    removedColumns.push(`${index}`)
+            if (d.displayOption) {
+                for (let key in d.displayOption) {
+                    if (context.hasOwnProperty(key)) {
+                        this.state.removedColumnsKey = context[key];
```

Comment:
this should do a this.setState not this.state.....

Reply:
removed